From 59b0595ac7543b8a7390911c904732d413f6e6c6 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 11 Feb 2016 23:21:31 -0500
Subject: [PATCH] Auto disable plugins that fail to load

Currently, a plugin that errors in onEnable is actually still
loaded and could be in a partial state...

Unload the plugin to ensure it does not cause undesired effects
if it fails to load correctly.
---
 src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java b/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
index a5f6cc1..35188a1 100644
--- a/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
+++ b/src/main/java/org/bukkit/plugin/java/JavaPluginLoader.java
@@ -335,6 +335,8 @@ public final class JavaPluginLoader implements PluginLoader {
                 jPlugin.setEnabled(true);
             } catch (Throwable ex) {
                 server.getLogger().log(Level.SEVERE, "Error occurred while enabling " + plugin.getDescription().getFullName() + " (Is it up to date?)", ex);
+                disablePlugin(jPlugin);
+                return;
             }
 
             // Perhaps abort here, rather than continue going, but as it stands,
-- 
2.7.1

